python/sparkdl/transformers/named_image.py

```diff
@@ -115,48 +120,65 @@ def _getIntermediateOutputCol(self):
         return "__tmp_" + self.getOutputCol()
 
 
-# TODO: give an option to take off multiple layers so it can be used in tuning
-#       (could be the name of the layer or int for how many to take off).
-class DeepImageFeaturizer(Transformer, HasInputCol, HasOutputCol):
+def _getScaleHintList():
+    featurizer = SparkContext.getOrCreate()._jvm.com.databricks.sparkdl.DeepImageFeaturizer
+    if isinstance(featurizer, py4j.java_gateway.JavaPackage):
+        # do not see DeepImageFeaturizer, possibly running without spark
+        # instead of failing return empty list
+        return []
+    return dict(featurizer.scaleHintsJava()).keys()
+
+
+class _LazyCaleHintConverter:
+    _sizeHintConverter = None
+
+    def __call__(self, value):
+        if not self._sizeHintConverter:
+            self._sizeHintConverter = SparkDLTypeConverters.buildSupportedItemConverter(
+                _getScaleHintList())
+        return self._sizeHintConverter(value)
+
+
+_scaleHintConverter = _LazyCaleHintConverter()
+
+
+class DeepImageFeaturizer(JavaTransformer, HasInputCol, HasOutputCol):
     """
     Applies the model specified by its popular name, with its prediction layer(s) chopped off,
     to the image column in DataFrame. The output is a MLlib Vector so that DeepImageFeaturizer
     can be used in a MLlib Pipeline.
-    The input image column should be 3-channel SpImage.
+    The input image column should be ImageSchema.
     """
 
     modelName = Param(Params._dummy(), "modelName", "A deep learning model name",
                       typeConverter=SparkDLTypeConverters.buildSupportedItemConverter(SUPPORTED_MODELS))
 
+    scaleHint = Param(Params._dummy(), "scaleHint", "Hint which algorhitm to use for image resizing",
+                      typeConverter=_scaleHintConverter)
+
```

Comment:
Did you want to add getter and setter for scaleHint & modelName.

Reply:
Yeah, already added.